sw/device/boot_rom/rom_crt.S

```diff
@@ -75,18 +81,39 @@ _reset_start:
 _start:
   .globl _start
 
+  // Enable entropy complex - this is not the full enable
+  li   a0, TOP_EARLGREY_ENTROPY_SRC_BASE_ADDR
+  li   t0, 0x2
+  sw   t0, ENTROPY_SRC_CONF_REG_OFFSET(a0)
+  li   a0, TOP_EARLGREY_CSRNG_BASE_ADDR
+  li   t0, 0x1
+  sw   t0, CSRNG_CTRL_REG_OFFSET(a0)
+  li   a0, TOP_EARLGREY_EDN0_BASE_ADDR
+  li   t0, 0x1
+  sw   t0, EDN_CTRL_REG_OFFSET(a0)
+
+  // Request memory scrambling and init
+  li   a0, TOP_EARLGREY_SRAM_CTRL_MAIN_BASE_ADDR
+  li   t0, 1
+  slli t1, t0, SRAM_CTRL_CTRL_RENEW_SCR_KEY_BIT
+  mv   a1, t1
+  slli t1, t0, SRAM_CTRL_CTRL_INIT_BIT
+  or   a1, a1, t1
```

Comment:
ah yeah that would be nice...

Reply:
As an aside I actually really like the `ROTATE THEN {AND|INSERT|OR|XOR} SELECTED BITS` in the IBM Z instruction set. Great for doing 'clever' things. Difficult to target in a compiler though since there is a lot of functional overlap with other instructions.